lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -106,10 +107,18 @@ def device_change attributes
         def boot_options attributes
           # NOTE: you must be using vsphere_rev 5.0 or greater to set boot_order
           # e.g. Fog::Compute.new(provider: "vsphere", vsphere_rev: "5.5", etc)
-          return unless @vsphere_rev.to_f >= 5
-          RbVmomi::VIM::VirtualMachineBootOptions.new(
-            :bootOrder => boot_order(attributes)
-          )
+          options = {}
+          if @vsphere_rev.to_f >= 5 and attributes[:boot_order]
+            options[:bootOrder] = boot_order(attributes)
+          end
+
+          # Set attributes[:boot_retry] to a delay in miliseconds to enable boot retries
+          if attributes[:boot_retry]
+            options[:bootRetryEnabled] = true
+            options[:bootRetryDelay]   = attributes[:boot_retry]
+          end
+                   
+          options.empty? ? nil : RbVmomi::VIM::VirtualMachineBootOptions.new(options)
```

Comment:
`options` is a bit of a poor name, but this is the constructed boot options, not the `attributes` that's passed into the method. So it is always empty on line 110 and then depending on the two following if clauses it might gain values.